### ISSUE TITLE
Vocore screen flickering 

### ISSUE THREAD

**Issue by FullPace** [SHWotever/SimHub#1584]
Hey bro! 
I found a very weird behaviour.

when using a formatting function on a quickly changing text element, like [currentlaptime] or delta times 
some vocore displays start to flicker (5%-10% fluctuation of brightness on the whole screen)

If I remove the formula, they act normal again. 
Here's what  i use as an example, nothing really fancy...

`if(timespantoseconds([CurrentLapTime])==0,'INLAP',
if (
timespantoseconds( [CurrentLapTime] ) < 600,
format([CurrentLapTime],'m\\:ss\\.f'),
format([CurrentLapTime],'mm\\:ss')
))`

With this there's constant flickering. As soon as I remove the function, it's normal again.
Can you think of anything causing this?

**Comment by SHWotever** [comment]
Hi ! Could you  join the full sample ? As is I highly doubt a single formula would be the cause.

**Comment by FullPace** [comment]
Here I send you the full file and the debugging dash (it's a blank template and this formula was enough to freak the screen out).

[Pacelogic Pro 2.5 beta.zip](https://github.com/SHWotever/SimHub/files/14842644/Pacelogic.Pro.2.5.beta.zip)

**Comment by SHWotever** [comment]
Just one question, are you using the mpro version ? Can you check the firmware is up to date ?

**Comment by FullPace** [comment]
here's a video of the flickering, so you have a better understanding. The camera does a good job here adjust the levels, in reality it's even more visible
https://photos.app.goo.gl/7ZBefRb1kuq8mAaf6

**Comment by SHWotever** [comment]
Unfortunately it's not possible to check the firmware directly. Non "mpro" version should be up to date.

Seeing the flash but the picture always right for me it's a backlight voltage irregularity emphasized by the very high contrast of your dash. I will check tommorow on my devices, but I fear this is an hardware quirks.

**Comment by FullPace** [comment]
I just checked firmware and it didn't install anything new. So it should be up to date.

The strange thing is it is working perfectly fine when the function is removed
https://photos.app.goo.gl/qLCLwBQzJ5YcAkYe7

**Comment by SHWotever** [comment]
Could you try with a closer case ? simply formatting use a single f (10th of seconds) In your comparaison you are comparing very quickly changing content with 3 seconds digits vs a single one, which probably helps to average the brightness.

**Comment by SHWotever** [comment]
Aso trying with significantly smaller font sizes would allow to verify my theory if it show no issues with the same content and smaller fonts that means it's hardware,

**Comment by FullPace** [comment]
sure, will do that tomorrow! I will also try adding a background color to lower the contrast.

**Comment by FullPace** [comment]
hi, pretty busy last days, I could only test two cases:

a) lowering contrast - no difference at all, even a bright grey with white text behaves the same (maybe a little less flickering)
b) formatting equally: you're right, when formatted equally it behaves the same.

will do more testing on the weekend

**Comment by FullPace** [comment]
some more tests:

font size isn't the solution. on my delta page font is 40px smaller and it stills flickers the screen.
I'm clueless. 
It's also not flickering all the time. on some replays I can reproduce it starts flickering at the exact same points in the replay and between it's fine.

**Comment by SHWotever** [comment]
That's very strange, I couldn't put my hand on a 4" (I'm sometimes disorganised) I know it was the weakest by far about power. And that white text with large font on black  make me think it will draw power in an irregular way depending of the actual characters being shown.

When I was saying reduce the contract, I was thinking about trying a gray font colour on a more gray background. So the power draw will be less impacted by each single character change.

Alternatively to see if it's an intermediate display error (which I doubt) you can lower the refresh rate in the settings, it goes as low as 5fps which will allow to visualise a display error if there are any.

If there is no error, do not hesitate to contact vocore for support,it looks like a power regulation issue. The v7b vocore is absolutely intolerant to communication errors (it shifts when there is any anomaly)

**Comment by FullPace** [comment]
I tested multiple colors on grey backgrounds - still flickering.

At this point I don't think it's an hardware issue anymore.
I was just driving AMS2 and loaded an old dashboard to my 5" Vocore (I have this for more than two years without any issues) and it started also flickering like crazy. On on left side of the screen. Even the background image was partially cut off. It's a very basic dashboard and it would be a very strange coincidence if the dashboard suddenly breaks and shows the same issue... wouldn't it?
Here's the dash: https://www.overtake.gg/downloads/lcd-dashbboard.52674/

here's a video showing the flickering:
https://photos.app.goo.gl/bGf5imhjLwMm31bE6

I hope this helps finding the issue.
I also texted the vocore guys and send them the link to this issue. maybe they get involved.

Cheers

**Comment by SHWotever** [comment]
This one feels really different indeed the text blinking is clearly visible, no doubts here it's not hardware ;) I will download it and try ;)

**Comment by FullPace** [comment]
it might look different on the video but in reality the flickering looks pretty much the same.

**Comment by SHWotever** [comment]
Hi ! I did try the dashboard I can see on your video part of the text and scale blink it's clearly a software issue.

Can you try in a browser to see if it behaves the same ? http://127.0.0.1:8888/Dash?v=1712939724560#LCD%20Dashboard 
There is probably a triggering condition I can't put my finger on yet.

**Comment by FullPace** [comment]
Hi, i justed on two computers and both behave perfectly fine. No issues in the browser.

**Comment by FullPace** [comment]
In Addition: I just tested on a second Vocore screen and it's also flickering there

**Comment by SHWotever** [comment]
By curiosity on the other computer did the vocore had the same issue ? I've found a 4 inch in stock and testing it with [overtake.gg/downloads/lcd-dashbboard.52674](https://www.overtake.gg/downloads/lcd-dashbboard.52674/) but it does not show the blinking. I'm not telling it does not exists, I'm just trying to find the triggering conditions :D

**Comment by FullPace** [comment]
yes. the other vocore was on the other computer and it had the same issue.

so far the lcd dashboard is flickering on all of my vocores.
the pacelogic dash only on one. I could "fix" by adding a bright white bar on top.
this prevents the flickering. but the white bar is not fitting on all devices, so that's not really the solution...

![image](https://github.com/SHWotever/SimHub/assets/116221480/221f50f2-8465-49a6-80ce-65769d51879d)

as soon as the white bar is a just a notch darker (like #fefefe ) the flickering comes back.

**Comment by FullPace** [comment]
i wish i could help more. I spent hours on this but I don't see whats triggering the behaviour

**Comment by raulramires** [comment]
Hello, I am having a similar issue. I designed a simple dashboard for WRC and it is noticeable there. Some things I have already tried:
- Increasing and decreasing the screen brightness and contrast;
- Decreasing the screen FPS;
- Changed the white color on the texts for a dark gray seems to improve a bit but still happens;
- Tried many different Dashboards;
- Tried different USB port and cable;

Nothing seems to improve. One thing I tried to do was take some screenshots of the dashboards and use them with the voCore test display software. I notice 0 brightness changes.
I sent an email to VoCore and they said that the screen has no auto brightness adjustment, which is probably a software issue.

I am sending my dashboard for anyone who wants to test, it is noticeable on the second one when the green bars light up.

The model of my screen is a 5-inch V7B D500FPC9373-C 854x480.

Any help would be appreciated. Thanks!

[WRC.zip](https://github.com/user-attachments/files/16168996/WRC.zip)
